api/desecapi/models.py

```diff
@@ -132,25 +131,33 @@ def pdns_sync(self):
         if new_domain:
             changes_required = bool(self.arecord) or bool(self.aaaarecord)
         else:
-            orig_domain = Domain.objects.get(id=self.id)
-            changes_required = self.arecord != orig_domain.arecord or self.aaaarecord != orig_domain.aaaarecord
+            changes_required = self.arecord != self._orig.arecord or self.aaaarecord != self._orig.aaaarecord
 
         # make changes if necessary
         if changes_required:
             pdns.set_dyn_records(self.name, self.arecord, self.aaaarecord)
 
+    @transaction.atomic
     def delete(self, *args, **kwargs):
+        super(Domain, self).delete(*args, **kwargs)
+
         pdns.delete_zone(self.name)
         if self.name.endswith('.dedyn.io'):
             pdns.set_rrset('dedyn.io', self.name, 'DS', '')
             pdns.set_rrset('dedyn.io', self.name, 'NS', '')
-        super(Domain, self).delete(*args, **kwargs)
 
+    @transaction.atomic
     def save(self, *args, **kwargs):
+        try:
+            self._orig = Domain.objects.get(pk=self.pk)
```

Comment:
use `__setattr__` to track changes https://stackoverflow.com/questions/2898711/django-model-fields-getter-setter